scripts/api.js

```diff
@@ -1,17 +1,40 @@
-/*global $ codeUrl */
-/* exported makeApiAjaxSettings */
+/*global codeUrl */
+/* exported ajax */
 
-function makeApiAjaxSettings(apiUrl, queryParams) {
-    let queryParamString = "";
-    if(queryParams) {
-        queryParamString = "&" + $.param(queryParams);
-    }
-    let settings = {
-        "url": codeUrl + "/api/index.php?url=" + apiUrl + queryParamString,
-        "dataType": "json",
-        "headers": {
-            "X-API-KEY": "SESSION"
-        }
+function ajax(method, apiUrl, queryParams = {}, data = {}) {
+    let url = new URL(`${codeUrl}/api/index.php`);
+    queryParams.url = apiUrl;
```

Comment:
Another problem with my way is that the url parameters get in the "wrong" order: the query parameters are before the "resource locator". This doesn't matter but if we make a "real" restful url we would have change this anyway. I'll make it so that the "url" part is before the actual query parameters. (This is not noticeable in the page browser because there are never any query parameters)